Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -335,6 +367,9 @@ extension _BSONDecoder {
         } else if type == UUID.self {
             // We know T is a UUID and unboxUUID returns a UUID or throws, so this cast will always work.
             return try unboxUUID(value) as! T
+        } else if type == Data.self {
```

Comment:
now that we are checking three different possible values of `type` here, would it make sense to rewrite this as a `switch`?